src/libcprover-cpp/api.h

```diff
@@ -79,11 +79,28 @@ struct api_sessiont
   /// Validate the loaded goto model
   void validate_goto_model() const;
 
-  // A simple API version information function.
+  /// A simple API version information function.
   std::unique_ptr<std::string> get_api_version() const;
 
+  /// Process the model by running symex and the decision procedure.
+  /// @return a `unique_ptr` to the `verification_resultt` summary.
+  std::unique_ptr<verification_resultt> run_verifier() const;
+
+  /// Read a goto-binary from a given filename.
+  /// @warning Will error out if it reads a source file.
+  void read_goto_binary(std::string &file) const;
+
+  /// True if file is goto-binary.
+  bool is_goto_binary(std::string &file) const;
```

Comment:
Maybe here,  -> `is_symex_ready_goto_binary`?

Reply:
We don't want to do this, because we delegate to `is_goto_binary` from `src/goto-programs/read_goto_binary.cpp` which doesn't check the "level" of the goto-binary (whether transformations have been performed and which ones), so we can't really promise to give information we don't have.